### PR TITLE
Auth: Add RemoteUser implementation

### PR DESCRIPTION
**Pull request by kiblik**
Implementation for https://owasp.slack.com/archives/C014H3ZV9U6/p1658840111091849

### CODE REVIEW

**Comment by kiblik** [comment]
@wurstbrot, can you check the changes of `dojo/pipeline.py`, please?

**Comment by wurstbrot** [comment]
nice refactoring @kiblik

File: dojo/remote_user.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class RemoteUserMiddleware(OriginalRemoteUserMiddleware):
+    def process_request(self, request):
+        # process only if request is comming from the trusted proxy node
+        if IPAddress(request.META['REMOTE_ADDR']) in settings.AUTH_REMOTEUSER_TRUSTED_PROXY:
+            self.header = settings.AUTH_REMOTEUSER_USERNAME_HEADER
+            return super().process_request(request)
+        else:
+            logger.debug('Requested came from untrusted proxy %s; This is list of trusted proxies: %s',
```
Comment (dsever) [inline dojo/remote_user.py:18]: I assume this should be error, not debug

Comment (kiblik) [reply dojo/remote_user.py:18]: No. This is only for debugging purpose if somebody's request is not processed as expected.
If the request is coming from different IP, RemoteUser middleware is simply ignored and continues with another middleware.
The authentication can be performed by other configured middleware like SAML or standard Username+Password

**Review by Maffooch — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Pull request merged**